compiler/rustc_feature/src/active.rs

```diff
@@ -43,7 +37,15 @@ macro_rules! declare_features {
             &[$(
                 // (sym::$feature, $ver, $issue, $edition, set!($feature))
                 Feature {
-                    state: State::Active { set: set!($feature) },
+                    state: State::Active {
+                        // Sets this feature's corresponding bool within `features`.
+                        set: {
+                            fn f(features: &mut Features) {
+                                features.$feature = true;
+                            }
+                            f as fn(&mut Features)
```

Comment:
Capture-less closures coerce into fn pointers, so you probably don't need a separate function or an explicit cast here.

Reply:
You are right that the cast can be removed. But I can't see how to remove the separate named function.

Reply:
Oh, use a closure, I see. Good suggestion.

Reply:
I added a new commit converting this to a closure. Thanks!